.ci/script_linux.sh

```diff
@@ -12,7 +12,11 @@ set -e
 ./scripts/internal-run.sh catkin build --no-status --no-deps -p 1 -i --cmake-args -DCMAKE_BUILD_TYPE=$BUILD_TYPE -DTREAT_WARNINGS_AS_ERRORS=ON -DCODECOV=$CODECOV --make-args tests -- aikido
 
 # Run tests and measure test coverage if CodeCov is on.
-if [ $CODECOV = ON ]; then ./scripts/internal-run.sh make -C build/aikido aikido_coverage; else ./scripts/internal-run.sh make -C build/aikido test; fi
+if [ $CODECOV = ON ]; then
+  ./scripts/internal-run.sh make -C build/aikido aikido_coverage
+else 
+  ./scripts/internal-run.sh cd build/aikido && ctest --output-on-failure && cd ../..
```

Comment:
Maybe instead of manually calling `ctest --output-on-failure`, we can set the `CTEST_OUTPUT_ON_FAILURE` environment variable?

Reply:
I think either way is possible. The actual problem here is that we should run `ctest` in the directory where the cmake is configured (i.e., `build/aikido`), but I don't know how to do that using `internal-run.sh`.

Reply:
Doesn't the `-C` flag to `make` already do that?

```
       -C dir, --directory=dir
            Change to directory dir before reading the makefiles or doing any-
            thing else.  If multiple -C options are specified, each is  inter-
            preted  relative to the previous one: -C / -C etc is equivalent to
            -C /etc.  This is typically used  with  recursive  invocations  of
            make.
```

Reply:
Yes, but `make` doesn't have `--output-on-failure` option. It seems none of `make test` and `ctest` has both of `-C` and `--output-on-failure` options.

Reply:
Right, that's why I was suggesting to set the environment variable instead. I think `CTEST_OUTPUT_ON_FAILURE=true make -C build/aikido test` should work.

Reply:
Hm, I presumed `CTEST_OUTPUT_ON_FAILURE` only works with `ctest` but not `make` since the environment begin with `CTEST_`. Let me try this though.

Reply:
`CTEST_OUTPUT_ON_FAILURE` doesn't work. :disappointed: Some tests are failing but the error messages are not printed.

Reply:
Hmm, this is what I see when I set `CTEST_OUTPUT_ON_FAILURE`: https://gist.github.com/brianhou/2207f29212ebceb06922d4f4cfbe08b1

I broke a test in `test_InstantaneousTrajectoryExecutor`, then ran `make -C build/aikido test` and `CTEST_OUTPUT_ON_FAILURE=true make -C build/aikido test`.

```
bhou1@r2:aikido-ws $ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 14.04.5 LTS
Release:	14.04
Codename:	trusty
bhou1@r2:aikido-ws $ ctest --version
ctest version 2.8.12.2
```

Reply:
Your test seems to show `CTEST_OUTPUT_ON_FAILURE=true` works. Could you please break a test so that it fails with an exception segfault like [this](https://ci.appveyor.com/project/jslee02/aikido/build/77/job/tjmni250aj1pkrkv#L5854)?

Reply:
[Yep](https://gist.github.com/brianhou/2207f29212ebceb06922d4f4cfbe08b1#file-gistfile2-txt).

Reply:
I think I figured it out: setting `CTEST_OUTPUT_ON_FAILURE` before calling `make -C build/aikido test` is different than `CTEST_OUTPUT_ON_FAILURE=true make -C build/aikido test`: https://gist.github.com/brianhou/2207f29212ebceb06922d4f4cfbe08b1#file-gistfile3-txt